src/main/java/com/openlattice/authorization/AceValue.java

```diff
@@ -23,24 +23,28 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import com.openlattice.authorization.securable.SecurableObjectType;
-import java.util.Collection;
-import java.util.EnumSet;
-import java.util.Iterator;
-import java.util.Set;
-import java.util.Spliterator;
+
+import java.util.*;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
 
 /**
  * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
  */
-public class    AceValue implements Set<Permission> {
+public class AceValue implements Set<Permission> {
     private final EnumSet<Permission> permissions;
     private       SecurableObjectType securableObjectType;
+    private       Optional<Date>      optionalExpirationDate;
```

Comment:
Switch to OffsetDateTime and make this not optional. This will make it easier to index the field. Make the two arg constructor initialize expirationDate to OffsetDateTime.MAX (correspond to Year 999999999). This will also simplify your serializers and have a clearer representation in the database as well.

Reply:
This won't cause backwards compatibility issues?